source-hubspot-native/tests/snapshots/source_hubspot_native_tests_test_snapshots__discover__stdout.json

```diff
@@ -747,10 +751,3088 @@
+              "title": "Row Id",
+              "type": "integer"
+            }
+          },
+          "required": [
+            "op"
+          ],
+          "title": "Meta",
+          "type": "object"
+        }
+      },
+      "additionalProperties": true,
+      "properties": {
+        "_meta": {
+          "allOf": [
+            {
+              "$ref": "#/$defs/Meta"
+            }
+          ],
+          "default": {
+            "op": "u",
+            "row_id": -1
+          },
+          "description": "Document metadata"
+        },
+        "id": {
+          "title": "Id",
+          "type": "integer"
+        },
+        "updatedAt": {
+          "format": "date-time",
+          "title": "Updatedat",
+          "type": "string"
+        },
+        "workflows": {
+          "default": [],
+          "items": {
+            "type": "integer"
+          },
+          "title": "Workflows",
+          "type": "array"
+        }
+      },
+      "required": [
+        "id",
+        "updatedAt"
+      ],
+      "title": "Workflows",
+      "type": "object",
+      "x-infer-schema": true
+    },
+    "key": [
+      "/id"
+    ]
+  },
+  {
+    "recommendedName": "properties",
+    "resourceConfig": {
+      "name": "properties",
+      "interval": "PT420S"
+    },
+    "documentSchema": {
+      "$defs": {
+        "Meta": {
+          "properties": {
+            "op": {
+              "description": "Operation type (c: Create, u: Update, d: Delete)",
+              "enum": [
+                "c",
+                "u",
+                "d"
+              ],
+              "title": "Op",
+              "type": "string"
+            },
+            "row_id": {
+              "default": -1,
+              "description": "Row ID of the Document, counting up from zero, or -1 if not known",
+              "title": "Row Id",
+              "type": "integer"
+            }
+          },
+          "required": [
+            "op"
+          ],
+          "title": "Meta",
+          "type": "object"
+        }
+      },
+      "additionalProperties": true,
+      "properties": {
+        "_meta": {
+          "allOf": [
+            {
+              "$ref": "#/$defs/Meta"
+            }
+          ],
+          "default": {
+            "op": "u",
+            "row_id": -1
+          },
+          "description": "Document metadata"
+        },
+        "name": {
+          "default": "",
+          "title": "Name",
+          "type": "string"
+        },
+        "calculated": {
+          "default": false,
+          "title": "Calculated",
+          "type": "boolean"
+        },
+        "hubspotObject": {
+          "default": "unknown",
+          "title": "Hubspotobject",
+          "type": "string"
+        }
+      },
+      "title": "Property",
+      "type": "object",
+      "x-infer-schema": true
+    },
+    "key": [
+      "/_meta/row_id"
```

Comment:
`_meta/row_id` needs to be a required property for this to work well with most materializations. So `_meta` needs to be required,  and in `_meta` `row_id` needs to be required.

Technically there is a default value for it set so it might work without this, but I think we might as well set the field as required since we know it will always be there.

Reply:
To solve the second requirement, i did a modification at estuary-cdk in this commit 344be6d90aa7b0f286f210242c7afa9f77c1279d

The first one ( add `_meta/row_id` ) seem a bit harder since its already required by the `BaseDocument` model, and i don't really know how to make both of then required since the models listed in models.py already require the BaseDocument model, could you help me to solve it?

Reply:
I don't think we want `row_id` to be required in `_meta` across the board for all connectors, since some of them won't have a concept of a `row_id` and will just use `_meta` to report `op`.

So, given that there is not a straightforward way to set this field as required and it _does_ have a default value which will allow it to be materialized (I actually tested this with our `source-google-sheets-native` connector, which has the same kind of schema), I'll retract my statement that this needs to be changed, and we should leave it alone for now and back out the change in https://github.com/estuary/connectors/commit/344be6d90aa7b0f286f210242c7afa9f77c1279d

Reply:
Got it!
Changes were reverted here 051e02e8056c6f40d56685a485d3ba9fc8674bed